### ISSUE TITLE
True division with integers causes GDFError

### ISSUE THREAD

**Issue by beckernick** [rapidsai/cudf#586]
## Reporting a bug

- [x] I am using the latest version built from `master`.
- [x] I have included the following environment details:
      Linux Distro, Linux Kernel, GPU Model
- [x] I have included the following version information for:
      Arrow, CUDA, Numpy, Pandas, Python
- [x] I have included below a minimal working reproducer (if you are unsure how
 to write one see http://matthewrocklin.com/blog/work/2018/02/28/minimal-bug-reports).

```
VERSION="16.04.5 LTS (Xenial Xerus)"
ID=ubuntu
ID_LIKE=debian
PRETTY_NAME="Ubuntu 16.04.5 LTS"
VERSION_ID="16.04"
HOME_URL="http://www.ubuntu.com/"
SUPPORT_URL="http://help.ubuntu.com/"
BUG_REPORT_URL="http://bugs.launchpad.net/ubuntu/"
VERSION_CODENAME=xenial
UBUNTU_CODENAME=xenial


Operating System:
VERSION="16.04.5 LTS (Xenial Xerus)"
ID=ubuntu
ID_LIKE=debian
PRETTY_NAME="Ubuntu 16.04.5 LTS"
VERSION_ID="16.04"

GPU:  Tesla V100

CUDA 9.2

Python 3.6.6

Arrow: N/A
pandas==0.20.3
numpy==1.14.3
```

While working with pygdf last night, standard true division on a `Series` of dtype `int`, `np.int32`, and `np.int64` unexpectedly resulted in a `GDFError: GDF_UNSUPPORTED_DTYPE`, with the example code and traceback shown below:

```
import numpy as np
import pandas as pd
from pygdf import Series, DataFrame

s = Series(pd.Series(np.random.sample(10), dtype=int))
s / 2
```
```
---------------------------------------------------------------------------
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/conda/envs/gdf/lib/python3.6/site-packages/pygdf-0.1.0a3+24.gc6ee1af-py3.6.egg/pygdf/series.py", line 294, in __truediv__
    return self._binaryop(other, 'truediv')
  File "/conda/envs/gdf/lib/python3.6/site-packages/pygdf-0.1.0a3+24.gc6ee1af-py3.6.egg/pygdf/series.py", line 269, in _binaryop
    outcol = self._column.binary_operator(fn, other._column)
  File "/conda/envs/gdf/lib/python3.6/site-packages/pygdf-0.1.0a3+24.gc6ee1af-py3.6.egg/pygdf/numerical.py", line 83, in binary_operator
    out_dtype=lhs.dtype)
  File "/conda/envs/gdf/lib/python3.6/site-packages/pygdf-0.1.0a3+24.gc6ee1af-py3.6.egg/pygdf/numerical.py", line 349, in numeric_column_binop
    null_count = _gdf.apply_binaryop(op, lhs, rhs, out)
  File "/conda/envs/gdf/lib/python3.6/site-packages/pygdf-0.1.0a3+24.gc6ee1af-py3.6.egg/pygdf/_gdf.py", line 71, in apply_binaryop
    binop(*args)
  File "/conda/envs/gdf/lib/python3.6/site-packages/libgdf_cffi-0.1.0a1-py3.6.egg/libgdf_cffi/wrapper.py", line 27, in wrap
libgdf_cffi.wrapper.GDFError: GDF_UNSUPPORTED_DTYPE
```

This error only appears to occur with `int` dtypes (both numpy as standard Python) and only with true division (addition/subtraction/multiplication/floor division still work). It appears to be coming from libgdf exposed through the wrapper in `def __getattr__` within  `libgdf`'s `python/libgdf_cffi/wrapper.py`.

@kkraus14 noticed that this appears to be due to libgdf only having elementwise division operations for float32 and float64 dtypes, as noted in: https://github.com/gpuopenanalytics/libgdf/blob/master/src/binaryops.cu#L229. Other binary ops such as floor division, subtraction, multiplication, currently support integer types.

Would be it possible to support additional types such as int32 and int64 for true division, as well?

**Comment by thomcom** [comment]
True division should just call floor division with integer types. Fix in PR #612.

**Comment by neerajvashistha** [comment]
can please update which all data-types are supported?
```
>>> import pandas as pd
>>> import cudf
>>> df = pd.read_csv('out_latest.csv', encoding = "ISO-8859-1")
>>> df.dtypes.to_dict()
{'text': dtype('O'), 'noun': dtype('O'), 'title_x': dtype('O'), 'price_list': dtype('O'), 'productid': dtype('O'), 'id_x': dtype('int64'), 'rating': dtype('float64')}
>>> gdf = cudf.read_csv('out_latest.csv', names = df.columns.tolist(),dtype = df.dtypes.to_dict())
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/conda/envs/rapids/lib/python3.5/site-packages/cudf-0+unknown-py3.5-linux-x86_64.egg/cudf/io/csv.py", line 131, in read_csv
    libgdf.read_csv(csv_reader)
  File "/conda/envs/rapids/lib/python3.5/site-packages/libgdf_cffi/wrapper.py", line 27, in wrap
    raise GDFError(errname, msg)
libgdf_cffi.wrapper.GDFError: GDF_UNSUPPORTED_DTYPE
>>> 
```
trying my hands on cudf, looks like a long way to go.

**Comment by thomcom** [comment]
Thanks for trying things out! Right now in particular we don't support strings (it is being worked on, and close to deployment). If you convert your strings and objects to either categorical or numerical types first, you should be able to get it all on the GPU. Strings soon!

**Comment by kkraus14** [comment]
Resolved by #612
